src/genome.py

```diff
@@ -109,8 +121,25 @@ def insert_te(self, pos: int, length: int) -> int:
 
         Returns a new ID for the transposable element.
         """
-        ...  # FIXME
-        return -1
+        if pos > len(self.lst):
+            pos = pos % len(self.lst)
```

Comment:
You don't need to check if the position is beyond the genome. If you take the remainder, it will always work. But nothing wrong with checking, of course.